src/develop/pixelpipe_hb.c

```diff
@@ -2134,6 +2134,23 @@ static int dt_dev_pixelpipe_process_rec_and_backcopy(dt_dev_pixelpipe_t *pipe, d
                                                      int pos)
 {
   dt_pthread_mutex_lock(&pipe->busy_mutex);
+  darktable.dtresources.level = 0;
```

Comment:
I think i know what you mean. But, this checking for level is intentionally done here to decouple ui changes from pipeline. Here we ensure that when processing the pipeline we have a correct parameter. We would otherwise also need a signal to reprocess the pipeline ... You don't agree?

Reply:
You mean if the pipeline is working while changing the preferences?

Reply:
yeah

Reply:
Ok, yet it is not good to have this done in each pipeline iteration, it is the case here, right?

So I still think we should use DT_SIGNAL_PREFERENCES_CHANGE and trigger a pipeline reprocess. The preference will be rarely changed, so better do nothing when it is not changed and do a somewhat larger work when it is changed.

Reply:
> it is the case here, right?

Yes :-( Your suggestion will also make further simplifications possible. Accepted !
Resolving this now, will do this tomorrow.